prediction_market_agent/db/models.py

```diff
@@ -74,6 +80,21 @@ class BlockchainMessage(SQLModel, table=True):
     consumer_address: str
     sender_address: str
     transaction_hash: str = Field(unique=True)
-    block: int = Field(sa_column=Column(BigInteger, nullable=False))
-    value_wei: int = Field(sa_column=Column(BigInteger, nullable=False))
+    block: int = Field(sa_column=Column(Numeric, nullable=False))
+    value_wei: int = Field(sa_column=Column(Numeric, nullable=False))
     data_field: Optional[str]
+
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def parsed_message(self) -> str:
+        """We try decompressing the message, else we return the original data field."""
+        if self.data_field is None:
+            return ""
```

Comment:
None and empty message are two different things, any reason to not return None here?

Reply:
returning `data_field` since it's already decompressed